FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,58 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case let .success((data, urlResponse)):
+				completion(FeedItemsMapper.decode(data, urlResponse))
+			}
+		}
+	}
+}
+
+// MARK: - Helpers
+
+private struct FeedItemsMapper {
+	private static var OK_200: Int {
+		200
+	}
+
+	private struct APIFeedImageResponse: Decodable {
+		let items: [APIFeedImage]
+	}
+
+	private struct APIFeedImage: Decodable {
+		let id: UUID
+		let description: String?
+		let location: String?
+		let url: URL
+
+		enum CodingKeys: String, CodingKey {
+			case id = "image_id"
+			case description = "image_desc"
+			case location = "image_loc"
+			case url = "image_url"
+		}
+
+		func mapToFeedImage() -> FeedImage {
+			FeedImage(id: id,
+			          description: description,
+			          location: location,
+			          url: url)
+		}
+	}
+
+	static func decode(_ data: Data, _ response: HTTPURLResponse) -> FeedLoader.Result {
```

Comment:
Since this method maps `data` and `response` to `FeedImages` you might want to name it `map`.
```suggestion
	static func map(_ data: Data, _ response: HTTPURLResponse) -> FeedLoader.Result {
```